src/mds/MDSRank.cc

```diff
@@ -2983,11 +2983,71 @@ bool MDSRankDispatcher::handle_command(
 
     damage_table.erase(id);
     return true;
+  } else if (prefix == "drop cache") {
+    command_drop_cache(ds);
+    return true;
   } else {
     return false;
   }
 }
 
+void MDSRank::command_drop_cache(stringstream *ds)
+{
+  if (not is_active()) {
+    dout(5) << __func__ << ": MDS not active, no-op" << dendl;
+    return;
+  }
+
+  if (not mdcache->trim(UINT64_MAX))
+    return;
+
+  server->recall_client_state(1, true);
+  mds_lock.Unlock();
+  if (not _wait_for_client_recall_state_acks())
+    dout(4) << "Timed out; didn't not receive "
+	    "CEPH_SESSION_RECALL_STATEMSG_ACK from all sessions." << dendl;
+
+  // Optionally, flush the journal -
+  Formatter *fmtr = new JSONFormatter(true);
+  command_flush_journal(fmtr);
+  // TODO: do we want output from `flush journal`?
```

Comment:
Running `ceph tell mds.a drop cache` displays output from `MDSRank::command_flush_journal()` as well as from `MDSRank::cache_status()`.
```
{
    "message": "",
    "return_code": 0
}
{
    "pool": {
        "items": 482,
        "bytes": 52398
    }
}
```
Do we want both or only the latter?

Reply:
both, combine into one object

Reply:
should be:

````
{
    "message": "",
    "return_code": 0
    "cache_status": {
      "pool": {
          "items": 482,
          "bytes": 52398
      }
  }
}
````

Reply:
Okay. Personally, I find it difficult to understand the meaning of first two lines. Will it be better if I make them a part of "flush journal" section like the following snippet?

```
{
    "flush_journal": {
        "message": "",
        "return_code": 0
    },
    "cache": {
        "pool": {
            "items": 482,
            "bytes": 52398
        }
    }
}
```

Reply:
Updating the PR with "flush_journal" part included. Let me know if you think it's not good idea.